components/script/dom/gpu.rs

```diff
@@ -40,7 +44,23 @@ impl GPU {
     }
 
     fn wgpu_channel(&self) -> Option<WebGPU> {
-        self.global().as_window().webgpu_channel()
+        let global = self.global();
+        let window = global.as_window();
+        if window.webgpu_channel().is_none() && !self.webgpu_channel_requested.get() {
+            self.webgpu_channel_requested.set(true);
+            let (sender, receiver) = ipc::channel().unwrap();
+            let script_to_constellation_chan = global.script_to_constellation_chan();
+            if script_to_constellation_chan
+                .send(ScriptMsg::StartWebGPU(sender))
```

Comment:
So I would say you could make the `WebGPURequest::RequestAdapter` into a `ScriptMsg`, and in `RequestAdapter` below just use the `script_to_constellation_chan` to send it to the constellation. 

Then the constellation can 

1. Start a webgpu if needed,
2. forward the `RequestAdapter` call.

And webgpu can then reply on a channel included in the `RequestAdapter` message with a `webgpu_channel` which can be stored in the adapter(so that it can communicate directly with webgpu going forward). 

The main benefit is not blocking on the constellation reply inside the `wgpu_channel` call, instead making it non-blocking as the actual `RequestAdapter` call, where the reply eventually resolves via a promise. 

You cold store the `webgpu_channel` somewhere besides the adapter, however it doesn't seem to have much benefit since any method of `GPU` would seem to have to go through the actual webgpu component. So it would only make sense to store it internally on the adapter, so that methods of it could send a message directly to webgpu without having to go through the constellation.